File: DAA/Insertion_Sort/insertionSort.c
#include <stdio.h>

void InsertionSort(int arr[], int n)
{
    for (int i = 1; i < n; i++)
    {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key)
        {
            arr[j + 1] = arr[j];
            j--;
        }

        arr[j + 1] = key;
    }
}

void display(int arr[], int n)
{
    for (int i = 0; i < n; i++)
    {
        printf("%d ", arr[i]);
    }
    
    printf("\n");
}

int main()
{
    int arr[5] = {5, 4, 3, 2, 1};
    InsertionSort(arr, 5);
    display(arr, 5);
}

File: DAA/Insertion_Sort/InsertionSort.py
def display(arr,n):
    for i in range(n):
        print(arr[i],end=" ")
    print()

def InsertionSort(arr,n):
    for i in range(n):
        key = arr[i]
        j=i-1
        while(j>=0 and arr[j]>key):
            arr[j+1]=arr[j]
            j-=1
        arr[j+1]=key

arr=[5,4,3,2,1]
InsertionSort(arr,5)
display(arr,5)